sms-android/snippet.html

```diff
@@ -0,0 +1,240 @@
+       /* IE8+ */
+       background-image: linear-gradient(to bottom, #bfc7cd, #9ca3aa);
+       /* the standard */
+       box-shadow: 0px 1px 0px 0px rgba(0, 0, 0, 0.2), inset 0px -1px 0px 0px rgba(0, 0, 0, 0.3);
+       color: #eee;
+   }
+
+   #firefox-android-sms-{{ snippet_id }} .button.arrow span {
+       display: block;
+       margin-right: -12px;
+       padding: 0 30px 0 0;
+       background: url(https://www.mozilla.org/media/img/sandstone/buttons/arrow-go-r.png) right center no-repeat;
+   }
+
+   #firefox-android-sms-{{ snippet_id }} #number {
+       height: 25px;
+       width: 180px;
+       background: #fff;
+       border-color: #b2b2b2;
+       border-style: solid;
+       border-width: 1px;
+       padding: 3.3333333333333335px 6.666666666666667px;
+       border-radius: 3px;
+       box-shadow: 0 1px rgba(255, 255, 255, 0.5);
+   }
+  </style>
+  <script type="text/javascript">
+   //<![CDATA[
+   (function() {
+       var snippet = document.getElementById('firefox-android-sms-{{ snippet_id }}');
+       snippet.addEventListener('show_snippet', function() {
+           var input = document.querySelector('#firefox-android-sms-{{ snippet_id }} #number');
+           input.addEventListener('input', function() {
+               var button = document.querySelector('#firefox-android-sms-{{ snippet_id }} #submit-button');
+               if (input.value.length >= 10) {
+                   button.disabled = false;
+                   button.classList.remove('insensitive');
+               }
+               else {
+                   button.disabled = true;
+                   button.classList.add('insensitive');
+               }
+           });
+           var submit = document.getElementById('submit-button');
+           var form = document.getElementById('snippet-{{ snippet_id }}-sms-send');
+           submit.addEventListener('click', function(event) {
+               event.preventDefault();
+               sendMetric('sms-subscribe', function() { form.submit(); });
+           });
+
+           // Logo change.
+           {% if replacement_logo %}
+             var newLogo = '{{ replacement_logo }}';
+             var brandLogo = document.getElementById('brandLogo');
+             if (brandLogo.src !== undefined) {
+                 brandLogo.src = newLogo;
+             } else {
+                 brandLogo.style.backgroundImage = 'url(' + newLogo + ')';
+             }
+             brandLogo.className += ' switch-{{ snippet_id }}';
+           {% endif %}
+       });
+   })();
+   //]]>
+  </script>
+
+  <form id="snippet-{{ snippet_id }}-sms-send" method="post" action="https://www.mozilla.org/en-US/firefox/sms/">
+    <img id="right-img" src="{{ image }}">
+    <h1>{{ header }}</h1>
+    <p id="text">{{ text|safe }}</p>
+
+    <p class="input">
+      <input type="tel"
+             maxlength="14"
+             id="number"
```

Comment:
Use a class instead of an ID for this in case multiple instances of this go out.